src/swaps/boltz.rs

```diff
@@ -1446,13 +1569,75 @@ pub struct GetFeeEstimationResponse {
     pub lbtc: f64,
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct CreateBolt12OfferRequest {
+    pub offer: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub url: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct UpdateBolt12OfferRequest {
+    pub offer: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub url: Option<String>,
```

Comment:
Deserves a commit that `None` will result in the Webhook URL being removed from the offer